modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2651,11 +2678,19 @@ def is_reduce_fn(fn, deep_level=0):
             is_reduce_fn(x) for x in agg_func.values()
         ):
             return self._groupby_dict_reduce(
-                by, axis, agg_func, agg_args, agg_kwargs, groupby_kwargs, drop
+                by, agg_func, axis, groupby_kwargs, agg_args, agg_kwargs, drop
             )
 
-        if callable(agg_func):
-            agg_func = wrap_udf_function(agg_func)
+        if isinstance(agg_func, dict):
+            assert (
+                how == "axis_wise"
+            ), f"Only 'axis_wise' aggregation is supported with dictionary functions, got: {how}"
+        else:
+            grp_agg_func = agg_func
+            method = GroupByDefault.get_aggregation_method(how)
+            agg_func = lambda grp, *args, **kwargs: method(  # noqa: E731
+                grp, grp_agg_func, *args, **kwargs
+            )
```

Comment:
why not use a [functools.partial](https://docs.python.org/3.7/library/functools.html#functools.partial)?

Reply:
used `functools.partial` at the last commit